gradle-enterprise-conventions-maven-extension/README.md

```diff
@@ -0,0 +1,53 @@
+# Example Gradle Enterprise Conventions Maven Extension
+
+This project demonstrates how to share a common Gradle Enterprise configuration across multiple Maven projects. It is intended to serve as a starting point for creating your own extension to apply your specific Gradle Enterprise configuration.
+
+## Usage
+
+* Create your own copy of the `extension` project
+* Address all `todo` comments
+    * Update the `groupId` and `artifactId` in the [POM](extension/pom.xml) file
+    * Update the package containing source files
+    * Update `isCi` in [CiUtils.java](extension/src/main/java/com/example/CiUtils.java) and the CI checks in the [Gradle Enterprise configuration](extension/.mvn/gradle-enterprise.xml) to properly detect your CI environment
+    * Update the server URL to your own Gradle Enterprise instance in the [Gradle Enterprise configuration](extension/.mvn/gradle-enterprise.xml) and [CustomGradleEnterpriseConfig.java](extension/src/main/java/com/example/CustomGradleEnterpriseConfig.java)
+* Add any common tags, links, or values to [CustomGradleEnterpriseConfig.java](extension/src/main/java/com/example/CustomGradleEnterpriseConfig.java)
+* Deploy the extension to your own artifact repository
```

Comment:
Can maven extensions also be fetched from custom artifact repositories? Or do they only fetch from Maven central? I don't know the answer here but I am wondering since maven extensions are loaded possibly before artifact repositories can be set.

Reply:
I ran a test where I cleared maven local, deployed to a non-maven local file repo, and it was able to resolve from there when I declared that directory as a repository.

It seems to respect the repository definitions in the POM file and resolve extensions from those. It's hinted at in [this documentation](https://maven.apache.org/configure.html#mvn-extensions-xml-file). I do think it would be good to be explicit about that in this README.

Reply:
We should do a full e2e test of this using a real repository. I wonder if we could publish to https://repo.grdev.net/ for a test. I have had one customer report problems with this. But I have another customer who hasn't.

Reply:
I originally verified this with an OSS repository manager, [Archiva](https://archiva.apache.org/), which appears to now be retired.

Reply:
I think we should be okay. After refreshing my memory on the reported issue, it's a result of not mirroring Maven Central. That's already mentioned in the README.